### ISSUE TITLE
JSON headers for CSV should be structured

### ISSUE THREAD

**Issue by jakebeal** [TASBE/TASBEFlowAnalytics#415]
The current JSON headers for CSV files are unstructured, in the sense that they do not include information about the nature of the variables that are being output.

Consider the use of `savejson` applied to structures (not objects), e.g.:

`savejson('',struct(struct(CM).Channels{1}))
`
produces:
```
{
	"name": "FITC-A",
	"Laser": 488,
	"FilterCenter": 515,
	"FilterWidth": 20,
	"PseudoUnits": 0,
	"Units": "ERF",
	"description": {
		"name": "FITC-A",
		"rawname": "FITC-A",
		"range": 262144,
		"bit": 32,
		"voltage": 180,
		"filter": null,
		"gain": 1,
		"emitter_wavelength": null,
		"emitter_power": null,
		"percent_light": null,
		"detector": null,
		"decade": 0,
		"log": 0,
		"logzero": 0
	},
	"LineSpec": "y",
	"PrintName": "EYFP",
	"unprocessed": 0
}

```

**Comment by coverney** [comment]
I turned it into a struct and it looks something like this:
```
{
	"channel_names": [
		"FITC-A",
		"PE-Tx-Red-YG-A",
		"Pacific Blue-A",
		"FSC-A"
	],
	"print_names": [
		"EYFP",
		"mKate",
		"EBFP2",
		"FSC"
	],
	"units": [
		"MEFL",
		"MEFL",
		"MEFL",
		"Eum"
	]
}
```
Is this structure okay?

**Comment by jakebeal** [comment]
This structure depends on maintaining the same order and number of elements in each subset, in order to implicitly group together `FITC-A`, `EYFP`, and `MEFL`.

I'd prefer to have a two-level structure that explicitly binds together the information, more like:
```
{
	"channels": [
		{
		"name":"FITC-A",
		"print_name":"EYFP",
		"units":"MEFL"
                },
		{
		"name":"PE-Tx-Red-YG-A",
		"print_name":"mKate",
		"units":"MEFL"
                }
	]
}
```

**Comment by coverney** [comment]
There is currently a chicken and the egg testing problem between the tutorial and the branch for this issue. It would be great if we could fulfill the tutorial pull request to test out the TASBE branch. It works locally on my machine.

**Comment by jakebeal** [comment]
You could also put a copy of the JSON in the tests directory of this project instead.

**Comment by coverney** [comment]
Would that be a temporary fix?

**Comment by jakebeal** [comment]
(it wouldn't replace the tutorial instance, but it can be used to avoid these chicken & egg testing problems, like we do with the color models now)

**Comment by jakebeal** [comment]
It can be long-term --- keeping a test-purposes JSON header in the tests directory is cheap.

**Comment by coverney** [comment]
Sounds good! I put one in the test directory and am currently debugging a csv read problem I found.

**Comment by jakebeal** [comment]
Fixed by #416
